src/ch00-00-introduction.md

```diff
@@ -1,191 +1,139 @@
-within files. For this, we’ll use many of the concepts we discussed in the
-previous chapters.
-
-Chapter 13 explores closures and iterators: features of Rust that come from
-functional programming languages. In Chapter 14, we’ll examine Cargo in more
-depth and talk about best practices for sharing your libraries with others.
-Chapter 15 discusses smart pointers that the standard library provides and the
-traits that enable their functionality.
-
-In Chapter 16, we’ll walk through different models of concurrent programming
-and talk about how Rust helps you to program in multiple threads fearlessly.
-Chapter 17 looks at how Rust idioms compare to object-oriented programming
-principles you might be familiar with.
-
-Chapter 18 is a reference on patterns and pattern matching, which are powerful
-ways of expressing ideas throughout Rust programs. Chapter 19 contains a
-smorgasbord of advanced topics of interest, including unsafe Rust, macros, and
-more about lifetimes, traits, types, functions, and closures.
-
-In Chapter 20, we’ll complete a project in which we’ll implement a low-level
-multithreaded web server!
-
-Finally, some appendices contain useful information about the language in a
-more reference-like format. Appendix A covers Rust’s keywords, Appendix B
-covers Rust’s operators and symbols, Appendix C covers derivable traits
-provided by the standard library, Appendix D covers some useful development
-tools, and Appendix E explains Rust editions.
-
-There is no wrong way to read this book: if you want to skip ahead, go for it!
-You might have to jump back to earlier chapters if you experience any
-confusion. But do whatever works for you.
+ยินดีต้อนรับสู่ _Rust Programming Language_, เนื้อหาในส่วนนี้เป็นเนื้อหาเบื้องต้นของหนังสือเล่มนี้เกี่ยวกับภาษา Rust
+Rust นั้นช่วยให้เราสามารถเขียน software ที่รวดเร็ว และน่าเชื่อถือได้มากขึ้น
+
+ในการออกแบบภาษาโปรแกรมมักต้องเลือกระหว่าง ความสะดวกในการใช้งานของผู้พัฒนา (High-level) หรือความสามารถในการจัดการกับ hardware (Low-level) ซึ่ง Rust นั้นสามารถข้ามผ่านความท้าทายนี้มาได้ โดยคงไว้ซึ่งความสามารถในการจัดการกับ hardware ที่ดี (เช่น การจัดการการใช้งาน memory) โดยไม่ต้องเผชิญกับความยุ่งยากในการพัฒนา เหมือนอย่างภาษา Low-level อื่นๆที่เคยมีมา
+
+## Rust เหมาะสำหรับใคร
+
+Rust เป็นภาษาที่เหมาะกับผู้คนหลายกลุ่ม ด้วยหลากหลายเหตุผลเราลองมาดูกันถึงเหตุผลสำคัญที่ว่า ทำไมภาษานี้ถึงเหมาะสม
+
+### ทีมนักพัฒนา
+
+Rust ได้รับการพิสูจน์มาว่าเหมาะสมกับการทำงานเป็นทีม ที่ประกอบด้วยนักพัฒนาที่มีความเชี่ยวชาญ หรือความเข้าใจต่อภาษาที่หลากหลาย
+ภาษา Low-level มักเกิด bug อยู่เรื่อยๆ ซึ่งในภาษาเหล่านั้นก็ป้องกันการเกิด bug ด้วยการทดสอบอย่างหนัก หรือการตรวจสอบ code ด้วยนักพัฒนาที่มีประสบการณ์มากกว่า
+แต่ใน Rust นั้น compiler ของภาษาจะทำหน้าที่เป็นเหมือนผู้รักษาประตูของ code ผ่านการ compile เพื่อป้องกัน bug ที่ซ่อนอยู่ภายใน เพราะฉะนั้นด้วยการทำงานควบคู่ไปด้วยของ compiler จึงทำให้ ทีมนักพัฒนาสามารถมุ่งความสนใจในการพัฒนา software และออกแบบ logic ภายใน program แทนที่จะไปสนใจ bug ที่น่าปวดหัว
```

Comment:
gatekeeper role should not translate literally to ผู้รักษาประตู the sentence

"In Rust, the compiler plays a gatekeeper role by refusing to
compile code with these elusive bugs, including concurrency bugs."

should translate to something like

"แต่ใน Rust นั้น compiler ของภาษาจะทำหน้าที่เป็นผู้ช่วยตรวจสอบความถูกต้องของ code ด้วยการไม่ยอมให้ code ที่มี bug ซึ่งโดยปกติอาจจะยากต่อการตรวจจับ รวมถึง bug ที่เกี่ยวข้องกับ concurrency สามารถผ่านการ compile ไปได้"